src/zenml/integrations/huggingface/materializers/huggingface_datasets_materializer.py

```diff
@@ -33,13 +42,38 @@
 DEFAULT_DATASET_DIR = "hf_datasets"
 
 
+def extract_repo_url(checksum_str: str) -> Optional[str]:
+    """Extracts the repo url from the checksum URL.
+
+    An example of a checksum_str is:
+    "hf://datasets/nyu-mll/glue@bcdcba79d07bc864c1c254ccfcedcce55bcc9a8c/mrpc/train-00000-of-00001.parquet"
+    and the expected output is "nyu-mll/glue".
+
+    Args:
+        checksum_str: The checksum_str to extract the dataset name from.
+
+    Returns:
+        Optional[str]: The extracted dataset name.
```

Comment:
```suggestion
        The extracted dataset name.
```

This is specified above as a type annotation.